homeassistant/components/plex/media_player.py

```diff
@@ -125,52 +159,45 @@ def _clear_media_details(self):
         # Clear library Name
         self._app_name = ""
 
-    def refresh(self, device, session):
+    def update(self):
         """Refresh key device data."""
         self._clear_media_details()
 
-        if session:  # Not being triggered by Chrome or FireTablet Plex App
-            self._session = session
-        if device:
-            self._device = device
+        self._available = self.device or self.session
+
+        if self.device:
             try:
-                device_url = self._device.url("/")
+                device_url = self.device.url("/")
             except plexapi.exceptions.BadRequest:
                 device_url = "127.0.0.1"
             if "127.0.0.1" in device_url:
-                self._device.proxyThroughServer()
-            self._session = None
-            self._machine_identifier = self._device.machineIdentifier
-            self._name = NAME_FORMAT.format(self._device.title or DEVICE_DEFAULT_NAME)
-            self._device_protocol_capabilities = self._device.protocolCapabilities
-
-            # set valid session, preferring device session
-            if self._device.machineIdentifier in self.plex_sessions:
-                self._session = self.plex_sessions.get(
-                    self._device.machineIdentifier, [None, None]
-                )[0]
-
-        if self._session:
+                self.device.proxyThroughServer()
+            self._name = NAME_FORMAT.format(self.device.title or DEVICE_DEFAULT_NAME)
+            self._device_protocol_capabilities = self.device.protocolCapabilities
+
+        if not self.session:
+            self.force_idle()
+        else:
             if (
-                self._device is not None
-                and self._device.machineIdentifier is not None
-                and self._session.players
+                self.device is not None
+                and self.device.machineIdentifier is not None
+                and self.session.players
             ):
                 self._is_player_available = True
                 self._player = [
                     p
-                    for p in self._session.players
-                    if p.machineIdentifier == self._device.machineIdentifier
+                    for p in self.session.players
```

Comment:
We can use `next` and a generator expression since we're only interested in the only match.
```py
self._player = next((
    p for p in self.session.players
    if p.machineIdentifier == self.device.machineIdentifier))
```

Reply:
Looking at it closer I think that `self.device` and `self._player` might point to the same thing. I'll see if there's some difference I'm not considering.

Reply:
Apparently there's a difference in the data returned by each, with the session's player data being richer.